prime-dotnet-webapi/Services/MetabaseService.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Text;
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.IdentityModel.Tokens;
+using System.IO;
+using System.Collections.Generic;
+
+namespace Prime.Services
+{
+    public class MetabaseService : IMetabaseService
+    {
+        public MetabaseService()
+        { }
+
+        public string BuildMetabaseEmbeddedString()
+        {
+            var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(PrimeEnvironment.MetabaseApi.Key));
+
+            var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256Signature);
+
+            var header = new JwtHeader(credentials);
+
+            // 10 minutes in the future represented by milliseconds from epoch
+            var expired = DateTime.Now.AddMinutes(10).Subtract(DateTime.MinValue.AddYears(1969)).TotalMilliseconds;
```

Comment:
I think OAuth wants seconds since Unix Epoc, not milliseconds. Also, .Net MinValue is not the same as the Unix epoch.
Not 100% sure what should be used here RE: DateTime vs DateTimeOffest (time is hard), but there is a DateTimeOffset.ToUnixTimeSeconds() method that we can use. So just

`DateTimeOffset.Now.AddMinutes(10).ToUnixTimeSeconds()`